common/nmea_parser.h

```diff
@@ -0,0 +1,213 @@
+        istringstream split(textToParse);
+        // separating the text to sentences
+        try {
+            if (textToParse == "") throw "Empty string to parse";
+
+            for (string each; getline(split, each, delimiter); sentences.push_back(each));    
+            // find the sentence with the id we want
+            int sentenceNumber= -1;
+            for (vector<int>::size_type i = 0; i < sentences.size(); i++) {
+                size_t found = sentences[i].find(NMEA_sentence_id);
+                if (found!= string::npos) {
+                    sentenceNumber = i;
+                    break;
+                }
+            }
+            if (sentenceNumber < 0) {
+                throw "[NMEAParser:parseNMEAstring()] : cannot find NMEA id";
+            }
+            auto f = sentences[sentenceNumber];
+            // separating the sentence to words 
+            istringstream splitWord(f);
+            for (string word; getline(splitWord, word, w_delimiter); words.push_back(word));
+
+        } catch(const char *error) {
+            throw error;
+        } catch(...) {}
+
+        return words;
+    }
+
+    static GPSQuality getGpsQuality(const int &qualityId) {     
+        GPSQuality quality;  
+        switch(qualityId) {
+            case 0 : quality = GPSQuality::INVALID; break;
+            case 1 : quality = GPSQuality::GPSFIX; break;
+            case 2 : quality = GPSQuality::DGPSFIX; break;
+            case 3 : quality = GPSQuality::PPSFIX; break;
+            case 4 : quality = GPSQuality::RTK; break;
+            case 5 : quality = GPSQuality::FRTK; break;
+        }
+        return quality;
+    }
+
+    // parsing the string time to numbers
+    static TimeStamp parseStringTime(std::string timeString) {
+        try {
+            if (timeString== "") throw "Empty string when reading time";
+            int hour = stoi(timeString.substr(0,2));
+            int min  = stoi(timeString.substr(2,2));
+            int sec  = stoi(timeString.substr(4,2));
+            int msc  = stoi(timeString.substr(7,2));
+            TimeStamp time(hour, min, sec, msc);    
+            return time;
+        } catch(const char *s) {
+            throw s;
+        }  catch(...) { }
+        
+        TimeStamp time(0,0,0,0);
+        return time;
+    }
+ 
+    public:
+
+    static GPSData parseNMEA(const std::string  &toParse) {
+        using namespace std;
+
+        degree_t            latitude, longitude;
+        arcminute_t         latitudeMinutes, longitudeMinutes; 
+        char                latDirection, longDirection;    
+        meter_t             altitude;         
+        meters_per_second_t velocity;
+        double              horizontalDilution;
+        int                 numberOfSatelites;
+        int                 gpsQualityIndicator;
+        GPSQuality          qualityOfGps;
+        GPSData             data;
+        
+        try {
+            if (toParse == "") throw "Emtpy serial output";
```

Comment:
*Really* minor thing, but FYI I think ``toParse.empty()`` is supposed to be the way to do this (so that you're not constructing another string object then doing a comparison)

Reply:
:+1: